insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -578,6 +578,9 @@ public int getBatteryLevel() {
 
     @NonNull @Override
     public PumpEnactResult deliverTreatment(DetailedBolusInfo detailedBolusInfo) {
+        if (detailedBolusInfo.insulin == 0 || detailedBolusInfo.carbs > 0) {
+            throw new IllegalArgumentException(detailedBolusInfo.toString());
```

Comment:
Hi,
During my tests on Insight Driver, I saw that `throw new IllegalArgumentException(detailedBolusInfo.toString());` occurs about 20 times per day (and each time AAPS restart...)
I added a log and I get
`Illegal argument : Insulin: 0.0 Carbs: 0.0` (Insight doesn't manage carbs and when Insulin equals 0 nothing is done in driver...)
=> I commented this line to avoid too many restart....

Reply:
The contract for `Pumpinterface.deliverTreatment` now specifies the the caller must ensure no requests are made to the pump which the pump cannot satisfy. The statement verifies that contract to pin down where the issue is. In this case the caller is not adhering to the contract. If you change the line to `throw new IllegalArgumentException(detailedBolusInfo.toString(), new Exception())` you'll get a stacktrace that shows who is making the call.
tl;dr: this is an error that needs to be fixed. Removing the line will likely just cause issues further down where they're harder to trace back.

Reply:
Ok, thanks for the explain

Reply:
See Stacktrace below    
```
java.lang.IllegalArgumentException: {"bolusType":"SMB","carbTime":0,"carbs":0.0,"carbsDuration":0,"deliverAtTheLatest":1619475827673,"eventType":"CORRECTION_BOLUS","insulin":0.0,"lastKnownBolusTime":1619286422354,"timestamp":1619475832231}
        at info.nightscout.androidaps.plugins.pump.insight.LocalInsightPlugin.deliverTreatment(LocalInsightPlugin.java:567)
        at info.nightscout.androidaps.queue.commands.CommandSMBBolus.execute(CommandSMBBolus.kt:30)
        at info.nightscout.androidaps.queue.QueueThread.run(QueueThread.kt:118)
```